WalletWasabi/Tor/Socks5/Pool/Identities/RandomIdentity.cs

```diff
@@ -0,0 +1,22 @@
+﻿using WalletWasabi.Crypto.Randomness;
+
+namespace WalletWasabi.Tor.Socks5.Pool.Identities
+{
+	public class RandomIdentity : IIdentity
+	{		
+		public RandomIdentity(bool canTorCircuitBeReused)
```

Comment:
To understand this, it's important to have a look here: https://github.com/zkSNACKs/WalletWasabi/pull/5751/files#diff-9cd1112fadcedb8cf4bcedf4090c4cd2e4c2e7af585830667c8572892f724212R68

To summarize it, there are three cases:

* There is default identity which is modeled using `DefaultIdentity` class which is instantiated once (singleton basically) (this corresponds to `isolateStream=false`)
* Then we want to have for each HTTP request a new identity, so we [instantiate](https://github.com/zkSNACKs/WalletWasabi/pull/5751/files#diff-9cd1112fadcedb8cf4bcedf4090c4cd2e4c2e7af585830667c8572892f724212R72) `RandomIdentity` when sending that HTTP request
   * In this case, we don't want the tor circuit to be re-used, so we set `canTorCircuitBeReused=false`. 
   * Also, `canTorCircuitBeReused=false` leads to disposal of `TorTcpConnection` which is a good thing, I think.
* And the last case is the one where you want to have `TorHttpClient` with one new random identity so that as long as you use that `TorHttpClient` instance, all you HTTP requests will use the same Tor identity. This is because of this [line](https://github.com/zkSNACKs/WalletWasabi/pull/5751/files#diff-9cd1112fadcedb8cf4bcedf4090c4cd2e4c2e7af585830667c8572892f724212R71).
    * In this case the tor circuit can be shared, so `canTorCircuitBeReused=true` here.

> If the Name is different there will be a new Tor circuit, won't it?

Yes, but please pay attention when exactly there is new name, that's the important thing here.



> What is the difference between, new Name and a new Tor circuit?

When you use new Name, there will be new Tor circuit. 


Anyway, this is a long comment but if you have a look at the first link in this comment, I think it will be clear immediately.

Reply:
I would isolate the 3 cases with classes to make it more obvious. 

- DefaultIdentity
- NewIdentity, RandomIdentity, OnceRandom
- VariableIdentitly

So `canTorCircuitBeReused` can be removed from the ctor, less confusion.

Reply:
These are also good names. 

- Mode.DefaultIdentity 
- Mode.SingleIdentityPerLifetime
- Mode.NewIdentityPerRequest

Reply:
> I would isolate the 3 cases with classes to make it more obvious.

I actually started with that model. :) Should be easy to switch back.

Reply:
I have updated the identity names.

Reply:
Can you switch back? I don't get the new names at all. We are always talking about Tor identities, so everyone in the team understands what is an identity.

Reply:
I think we should start with names here as it seems that I use "identity" in a different way than you guys do and that brings the confusion. So in this PR, "identity" is simply a Tor circuit and then:

* OneOffIdentity - For case when you want to have a new Tor circuit for a new HTTP request that you don't want to link to any other previous one (not sure what scenario this covers but it's very expensive in terms of Tor's resources because a new circuit must be created always). 
* PersonIdentity - Supposed to be used for several HTTP requests which can be traced by an adversary to a single person (should model Alices and Bobs well (AFAIK)).
* DefaultIdentity - For downloading fee rates etc where privacy is not important really and it does not matter that anyone will find out that it is really you who downloaded fee rates 10 times this day (example).

Now, could you tell me if this API is actually useful for you and could you propose modifications to this API? 

I can revert back but when I don't understand the issue here fully, I can't say whether it makes sense or propose a better solution.

Maybe I can just change `Identity` to `Circuit` so that it does not confuse people.

Reply:
@molnard Do you find the latest version better?

Reply:
Yes